src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -915,9 +918,13 @@ protected void VerifyResponseContentLength()
                     _keepAlive = false;
                 }
 
-                ReportApplicationError(new InvalidOperationException(
-                    CoreStrings.FormatTooFewBytesWritten(_responseBytesWritten, responseHeaders.ContentLength.Value)));
+                ex = new InvalidOperationException(
+                    CoreStrings.FormatTooFewBytesWritten(_responseBytesWritten, responseHeaders.ContentLength.Value));
+                return false;
             }
+
+            ex = null;
+            return true;
```

Comment:
You could just return the Exception normally. Null could mean false.

Reply:
The try works better for the callers.